Editor/Scripts/CodeGenerators/GenerateRGStateClasses.cs

```diff
@@ -0,0 +1,276 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using UnityEditor;
+using UnityEngine;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace RegressionGames
+{
+    public static class GenerateRGStateClasses
+    {
+        public static void Generate(string jsonInput)
+        {
+            var rgStateInfos = JsonConvert.DeserializeObject<RGStateInfoWrapper>(jsonInput).RGStateInfo;
+
+            foreach (var rgStateInfo in rgStateInfos)
+            {
+                var className = $"{rgStateInfo.Object}_RGState";
+                var componentType = rgStateInfo.Object;
+
+                // Create a new compilation unit
+                var compilationUnit = SyntaxFactory.CompilationUnit()
+                    .AddUsings(SyntaxFactory.UsingDirective(SyntaxFactory.ParseName("System")),
+                        SyntaxFactory.UsingDirective(SyntaxFactory.ParseName("System.Collections.Generic")));
+
+                // Create a new class declaration with the desired name
+                var classDeclaration = SyntaxFactory.ClassDeclaration(className)
+                    .AddModifiers(SyntaxFactory.Token(SyntaxKind.PublicKeyword))
+                    .AddBaseListTypes(SyntaxFactory.SimpleBaseType(SyntaxFactory.ParseTypeName("RGState")));
+
+                // Create the private field
+                var fieldDeclaration = SyntaxFactory.FieldDeclaration(
+                        SyntaxFactory.VariableDeclaration(SyntaxFactory.ParseTypeName(componentType))
+                            .AddVariables(SyntaxFactory.VariableDeclarator("myComponent")))
+                    .AddModifiers(SyntaxFactory.Token(SyntaxKind.PrivateKeyword));
+
+                // Create the Start method
+                var startMethod = GenerateStartMethod(componentType, rgStateInfo.State);
+
+                // Create the GetState method
+                var getStateMethod = GenerateGetStateMethod(componentType, rgStateInfo.State);
+
+                // Add the members to the class declaration
+                classDeclaration = classDeclaration
+                    .AddMembers(fieldDeclaration, startMethod, getStateMethod);
+
+                // Create namespace
+                var namespaceDeclaration = SyntaxFactory.NamespaceDeclaration(SyntaxFactory.ParseName("RegressionGames.RGBotConfigs"))
+                                                       .AddMembers(classDeclaration);
+
+                // Add the namespace declaration to the compilation unit
+                compilationUnit = compilationUnit.AddMembers(namespaceDeclaration);
+
+                // Get the full code text
+                var formattedCode = compilationUnit.NormalizeWhitespace().ToFullString();
+
+                // Write the code to a .cs file
+                string headerComment = "/*\n* This file has been automatically generated. Do not modify.\n*/\n\n";
+
+                // Save to 'Assets/RGScripts/RGSerialization.cs'
```

Comment:
Should this say `"Save to 'Assets/RGScripts/RGStates/{name}.cs'"`

Reply:
Fixed